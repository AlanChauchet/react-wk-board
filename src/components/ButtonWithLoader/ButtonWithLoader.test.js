import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';

import { ButtonWithLoader } from './index';

it('renders a button which can load', () => {
  const button = mount(
    <ButtonWithLoader loading={false}>Test</ButtonWithLoader>
  );
  expect(button.prop('loading')).toBeFalsy();
  expect(button.find(CircularProgress).exists()).toEqual(false);
  expect(button).toMatchSnapshot('not loading state');
  button.setProps({
    loading: true,
  });
  expect(button.prop('loading')).toBeTruthy();
  expect(button).toMatchSnapshot('loading state');
  expect(button.find(CircularProgress).exists()).toEqual(true);
  button.unmount();
});
