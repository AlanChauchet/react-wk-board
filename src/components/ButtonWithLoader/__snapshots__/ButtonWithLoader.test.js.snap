// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button which can load: loading state 1`] = `
<WithStyles(_ButtonWithLoader)
  loading={true}
>
  <_ButtonWithLoader
    classes={
      Object {
        "buttonProgress": "_ButtonWithLoader-buttonProgress-2",
        "wrapper": "_ButtonWithLoader-wrapper-1",
      }
    }
    loading={true}
  >
    <div
      className="_ButtonWithLoader-wrapper-1"
    >
      <WithStyles(Button)
        disabled={true}
      >
        <Button
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-11",
              "contained": "MuiButton-contained-12",
              "containedPrimary": "MuiButton-containedPrimary-13",
              "containedSecondary": "MuiButton-containedSecondary-14",
              "disabled": "MuiButton-disabled-19",
              "fab": "MuiButton-fab-20",
              "flat": "MuiButton-flat-7",
              "flatPrimary": "MuiButton-flatPrimary-8",
              "flatSecondary": "MuiButton-flatSecondary-9",
              "focusVisible": "MuiButton-focusVisible-18",
              "fullWidth": "MuiButton-fullWidth-24",
              "label": "MuiButton-label-4",
              "mini": "MuiButton-mini-21",
              "outlined": "MuiButton-outlined-10",
              "raised": "MuiButton-raised-15",
              "raisedPrimary": "MuiButton-raisedPrimary-16",
              "raisedSecondary": "MuiButton-raisedSecondary-17",
              "root": "MuiButton-root-3",
              "sizeLarge": "MuiButton-sizeLarge-23",
              "sizeSmall": "MuiButton-sizeSmall-22",
              "textPrimary": "MuiButton-textPrimary-5",
              "textSecondary": "MuiButton-textSecondary-6",
            }
          }
          color="default"
          component="button"
          disableFocusRipple={false}
          disabled={true}
          fullWidth={false}
          mini={false}
          size="medium"
          type="button"
          variant="text"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-3 MuiButton-flat-7 MuiButton-disabled-19"
            component="button"
            disabled={true}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-18"
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-3 MuiButton-flat-7 MuiButton-disabled-19"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-26",
                  "focusVisible": "MuiButtonBase-focusVisible-27",
                  "root": "MuiButtonBase-root-25",
                }
              }
              component="button"
              disableRipple={false}
              disableTouchRipple={false}
              disabled={true}
              focusRipple={true}
              focusVisibleClassName="MuiButton-focusVisible-18"
              tabIndex="0"
              type="button"
            >
              <button
                className="MuiButtonBase-root-25 MuiButtonBase-disabled-26 MuiButton-root-3 MuiButton-flat-7 MuiButton-disabled-19"
                disabled={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="MuiButton-label-4"
                />
              </button>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
      <WithStyles(CircularProgress)
        className="_ButtonWithLoader-buttonProgress-2"
        size={24}
      >
        <CircularProgress
          className="_ButtonWithLoader-buttonProgress-2"
          classes={
            Object {
              "circle": "MuiCircularProgress-circle-41",
              "circleIndeterminate": "MuiCircularProgress-circleIndeterminate-43",
              "circleStatic": "MuiCircularProgress-circleStatic-42",
              "colorPrimary": "MuiCircularProgress-colorPrimary-38",
              "colorSecondary": "MuiCircularProgress-colorSecondary-39",
              "indeterminate": "MuiCircularProgress-indeterminate-37",
              "root": "MuiCircularProgress-root-35",
              "static": "MuiCircularProgress-static-36",
              "svg": "MuiCircularProgress-svg-40",
            }
          }
          color="primary"
          size={24}
          thickness={3.6}
          value={0}
          variant="indeterminate"
        >
          <div
            className="MuiCircularProgress-root-35 MuiCircularProgress-colorPrimary-38 MuiCircularProgress-indeterminate-37 _ButtonWithLoader-buttonProgress-2"
            role="progressbar"
            style={
              Object {
                "height": 24,
                "width": 24,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg-40"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle-41 MuiCircularProgress-circleIndeterminate-43"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </CircularProgress>
      </WithStyles(CircularProgress)>
    </div>
  </_ButtonWithLoader>
</WithStyles(_ButtonWithLoader)>
`;

exports[`renders a button which can load: not loading state 1`] = `
<WithStyles(_ButtonWithLoader)
  loading={false}
>
  <_ButtonWithLoader
    classes={
      Object {
        "buttonProgress": "_ButtonWithLoader-buttonProgress-2",
        "wrapper": "_ButtonWithLoader-wrapper-1",
      }
    }
    loading={false}
  >
    <div
      className="_ButtonWithLoader-wrapper-1"
    >
      <WithStyles(Button)
        disabled={false}
      >
        <Button
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-11",
              "contained": "MuiButton-contained-12",
              "containedPrimary": "MuiButton-containedPrimary-13",
              "containedSecondary": "MuiButton-containedSecondary-14",
              "disabled": "MuiButton-disabled-19",
              "fab": "MuiButton-fab-20",
              "flat": "MuiButton-flat-7",
              "flatPrimary": "MuiButton-flatPrimary-8",
              "flatSecondary": "MuiButton-flatSecondary-9",
              "focusVisible": "MuiButton-focusVisible-18",
              "fullWidth": "MuiButton-fullWidth-24",
              "label": "MuiButton-label-4",
              "mini": "MuiButton-mini-21",
              "outlined": "MuiButton-outlined-10",
              "raised": "MuiButton-raised-15",
              "raisedPrimary": "MuiButton-raisedPrimary-16",
              "raisedSecondary": "MuiButton-raisedSecondary-17",
              "root": "MuiButton-root-3",
              "sizeLarge": "MuiButton-sizeLarge-23",
              "sizeSmall": "MuiButton-sizeSmall-22",
              "textPrimary": "MuiButton-textPrimary-5",
              "textSecondary": "MuiButton-textSecondary-6",
            }
          }
          color="default"
          component="button"
          disableFocusRipple={false}
          disabled={false}
          fullWidth={false}
          mini={false}
          size="medium"
          type="button"
          variant="text"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-3 MuiButton-flat-7"
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-18"
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-3 MuiButton-flat-7"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-26",
                  "focusVisible": "MuiButtonBase-focusVisible-27",
                  "root": "MuiButtonBase-root-25",
                }
              }
              component="button"
              disableRipple={false}
              disableTouchRipple={false}
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="MuiButton-focusVisible-18"
              tabIndex="0"
              type="button"
            >
              <button
                className="MuiButtonBase-root-25 MuiButton-root-3 MuiButton-flat-7"
                disabled={false}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-4"
                >
                  Test
                </span>
                <WithStyles(TouchRipple)
                  center={false}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child-32",
                        "childLeaving": "MuiTouchRipple-childLeaving-33",
                        "childPulsate": "MuiTouchRipple-childPulsate-34",
                        "ripple": "MuiTouchRipple-ripple-29",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-31",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-30",
                        "root": "MuiTouchRipple-root-28",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-28"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-28"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </button>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
    </div>
  </_ButtonWithLoader>
</WithStyles(_ButtonWithLoader)>
`;
